File: packages/emitter/testcases/common/functions/optional-callbacks/OptionalParams.ts
import { int } from "@tsonic/core/types.js";

export type Callback = (result: int) => void;

// Optional callback parameter
export function compute(value: int, callback?: Callback): int {
  const result = value * 2;
  if (callback !== undefined) {
    callback(result);
  }
  return result;
}

// Nullable function type
export function maybeTransform(
  value: int,
  transform: ((x: int) => int) | null
): int {
  if (transform !== null) {
    return transform(value);
  }
  return value;
}

File: packages/emitter/testcases/common/expected/functions/optional-callbacks/OptionalParams.cs
namespace TestCases.common.functions.optionalcallbacks
{
    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class OptionalParams
    {
        public static int compute(int value, global::System.Action<int>? callback = default)
        {
            var result = value * 2;
            if (callback != null)
            {
                callback(result);
            }
            return result;
        }

        public static int maybeTransform(int value, global::System.Func<int, int>? transform)
        {
            if (transform != null)
            {
                return transform(value);
            }
            return value;
        }
    }
}